### PR TITLE
Add documentation for Observability

### PR DESCRIPTION
**Pull request by artembilan**
* Adapt Observation code to the latest dependencies
* Add doc generation tasks for meters and spans
* Document new Observation API features
* Include generated meters and spans docs to a general `metrics.adoc` chapter

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -84,8 +87,8 @@ ext {
     lettuceVersion = '6.2.0.RELEASE'
     log4jVersion = '2.19.0'
     mailVersion = '2.0.1'
-	micrometerVersion = '1.10.0-M6'
-    micrometerTracingVersion = '1.0.0-M8'
+	micrometerVersion = '1.10.0-SNAPSHOT'
```
Comment (onobc) [inline build.gradle:90]: **[TinyNit]** indentation

**Review by onobc — APPROVED** [review]
@artembilan 

I am glad I saw this PR. It has a couple of things I can leverage in `spring-pulsar` - thanks for that.

1. The removal of package tangles
2. The package/name reduction to `o.s.i` for the generated docs

LGTM.

File: build.gradle (hunk 1 above)
Comment (artembilan) [inline build.gradle:90]:
```suggestion
    micrometerVersion = '1.10.0-SNAPSHOT'
```

File: src/reference/asciidoc/whats-new.adoc [hunk 2]
```diff
@@ -56,6 +56,12 @@ See <<./amqp.adoc#rmq-streams,RabbitMQ Stream Queue Support>> for more informati
 The SFTP modules has been fully reworked from outdated JCraft JSch library to more robust and modern `org.apache.sshd:sshd-sftp` module of the Apache MINA project.
 
 See <<./sftp.adoc#sftp,SFTP Adapters>> for more information.
+
+[[x6.0-micrometer-observation]]
+==== Micrometer Observation
+
+Enabling observation for timers and tracing using Micrometer is now supported.
+See <<./metrics.adoc#micrometer-observation,Micrometer Observation>> for more information.
```
Comment (artembilan) [inline src/reference/asciidoc/whats-new.adoc:64]:
```suggestion
See <<./metrics.adoc#micrometer-observation,Micrometer Observation>> for more information.

```

**Pull request merged**